homeassistant/components/cast/home_assistant_cast.py

```diff
@@ -40,10 +40,20 @@
 
     async def handle_show_view(call: core.ServiceCall):
         """Handle a Show View service call."""
+        hass_url = hass.config.api.base_url
+
+        # Home Assistant Cast only works with https urls. If user has no configured
+        # base url, use their remote url.
+        if not hass_url.lower().startswith("https://"):
+            try:
+                hass_url = hass.components.cloud.async_remote_ui_url()
+            except hass.components.cloud.CloudNotAvailable:
+                pass
```

Comment:
What will happen below in this case?

Reply:
Home Assistant Cast will try to connect but fail because it's not using https. This error is printed to the screen.